client/src/components/form/Form.js

```diff
@@ -6,61 +6,86 @@ import Button from "./Button";
 import DropDown from "./DropDown";
 import { postTodo } from "../../api/postToDo";
 import DeleteButton from "../DeleteButton";
+import { deleteTodo } from "../../api/deleteToDo";
 
-const Form = ({ topic, placeholder, category }) => {
+const Form = ({
+  topic,
+  placeholder,
+  toDoId,
+  toDo,
+  onSetShowModal,
+  onRefetch,
+}) => {
   const [title, setTitle] = useState("");
   const [description, setDescription] = useState("");
+  const [category, setCategory] = useState("");
+  // const [loading, setLoading] = useState(false);
 
   async function handleSubmit(event) {
     event.preventDefault();
     const toDo = { title, category, description };
-
+    // setLoading(true);
     await postTodo(toDo);
+    onSetShowModal(false);
+    await onRefetch();
+  }
+
+  function handleCategoryChange(category) {
+    setCategory(category.value);
+  }
+
+  async function handleDeleteToDo(event) {
+    event.preventDefault();
+    await deleteTodo(toDo.id);
+    await onRefetch();
   }
+
   return (
-    <FormContainer>
+    <StyledForm onSubmit={handleSubmit}>
       <Input
-        onSubmit={handleSubmit}
         value={title}
         topic="Titel"
-        placeholder="neues ToDo"
+        placeholder={toDo ? toDo.title : "neues ToDo"}
         onChange={(event) => {
           setTitle(event.target.value);
         }}
       />
       <Input
         value={description}
         topic="Beschreibung"
-        placeholder="ToDo Beschreibung"
+        placeholder={toDo ? toDo.description : "ToDo Beschreibung"}
         onChange={(event) => {
           setDescription(event.target.value);
         }}
       />
-      <DropDown />
-      <DeleteButton />
+      <DropDown
+        category={category}
+        handleCategoryChange={handleCategoryChange}
+      />
+      <DeleteButton type="button" onClick={handleDeleteToDo} />
       <Button />
-    </FormContainer>
+    </StyledForm>
   );
 };
 
 export default Form;
 
 //styling
 
-const FormContainer = styled.form`
+const StyledForm = styled.form`
   display: flex;
   flex-direction: column;
-  /* align-items: center; */
-  /* justify-content: space-around; */
-  /* margin: 2em; */
+  background-color: #cebebe;
+  z-index: 2;
```

Comment:
Just a hint: A value of e.g. 200 would be better, if later you decide to have something between z-index: 1 and 'your 2', you would have 198 options, but like this none ;)